### ISSUE TITLE
Pre-existing vault file no longer accessible from Google Drive

### ISSUE THREAD

**Issue by kolole** [buttercup/buttercup-mobile#386]
Bcup version: 2.6.3
Android version: 14
Phone: Google Pixel 6a

**First sign of problem**
The vault file from Google Drive that was already set (and which was working fine up until recently) could not be opened. The error message could not be read fully, but was repeating, something like "Error, could not fetch: Error could not fetch: Auth Error"

**Second sign of problem**
After first sign was repeated a bunch of times by unsuccessfully trying to unlock the vault, I deleted all the app data, and tried adding the vault file again. After successfully linking Bcup up to my Google Drive, Bcup can no longer see the vault file that I have indeed saved on the root directory of Google Drive. 

**Troubleshooting**

- I thought that maybe the file had become corrupted, so I downloaded the vault file and tried opening it on my buttercup-desktop (linux version). That was successful - nothing seemingly wrong with the vault file. 

- In the mobile app, I tried to create a _new_  test vault, which was successful, and the vault file was also saved on the Google Drive.

- I've also tried to rename the original vault file from within Google Drive, to see if that fixed the issue. No joy.

- I uploaded a vault file I use on my laptop to Google Drive, but that file can't be seen by Bcup either

**Conclusion**
Nothing I've tried works. The only way for Bcup to see a vault file located on Google Drive, is to create a new vault from the mobile app - existing vault files on GD are not seen by Bcup Android.

**Comment by kolole** [comment]
I see that on Google Drive, Buttercup has the following properties under Security>Your connections to third party apps and services:

Buttercup has some access to your Google Account

Should Buttercup rather be asking to get _full_ access or something? That seems to be right, because the description to "some access" is that the app only has access to the files that were created by the app - but I'm essentially asking it to use files that were not created by this app, at least not this instance of it. I can't change that option, so I think this has to do with what the Buttercup app is requesting to Google.

**Comment by kolole** [comment]
This image shows that I have .bcup files on my Google Drive (which the mobile app cannot see)
![image](https://github.com/user-attachments/assets/a06fe07e-71d3-465c-88b2-0e3c3c82dd0f)

This image shows that bcup mobile cannot see the files
![image](https://github.com/user-attachments/assets/9c0cfdf1-8c71-44bd-9bd7-757d7ee61e78)

This image shows that Buttercup has _some_ (emphasis me) access to my Google Drive
![image](https://github.com/user-attachments/assets/132ce83d-b272-448f-b199-810f46460c44)

**Comment by Traimb0t** [comment]
I'm also experiencing this issue and buttercup has become unusable on Android.

**Comment by vpelaez** [comment]
This is also happening to me in iOS

**Comment by kolole** [comment]
Solved (for now) by using BitWarden instead. However, I'd really like to use BC, because it works so nicely on desktop and self-hosting the vault file on my server. I also created an issue to address the fact that BC-mobile can't read local files (which would be a link to my server using SMB), only store locally by creating a new vault.

BC mobile in short, doesn't seem to able to read files anymore. I could never get WebDav to work either.

**Comment by tehsphinx** [comment]
Yes, the access was changed to only allow access to files buttercup itself created. This made it impossible to add existing vault files on Google Drive.

Since that change I had to use Google Drive app to sync the file to a local folder on Desktop to make it work with desktop apps. Haven't found a workaround for iOS. Haven't been able to access the locally synched file from buttercup app on iOS.
